stubs/cffi/_cffi_backend.pyi

```diff
@@ -0,0 +1,290 @@
+import sys
+import types
+from _typeshed import Incomplete, ReadableBuffer, WriteableBuffer
+from collections.abc import Callable, Hashable
+from typing import Any, ClassVar, Protocol, TypeVar, overload
+from typing_extensions import Literal, TypeAlias, final
+
+_T = TypeVar("_T")
+
+class _Allocator(Protocol):
+    def __call__(self, cdecl: str | CType, init: Any = ...) -> _CDataBase: ...
+
+FFI_CDECL: int
+FFI_DEFAULT_ABI: int
+RTLD_DEEPBIND: int
+RTLD_GLOBAL: int
+RTLD_LAZY: int
+RTLD_LOCAL: int
+RTLD_NODELETE: int
+RTLD_NOLOAD: int
+RTLD_NOW: int
+
+@final
+class CField:
+    bitshift: Incomplete
+    bitsize: Incomplete
+    flags: Incomplete
+    offset: Incomplete
+    type: Incomplete
+
+@final
+class CLibrary:
+    def close_lib(self, *args, **kwargs): ...
+    def load_function(self, *args, **kwargs): ...
+    def read_variable(self, *args, **kwargs): ...
+    def write_variable(self, *args, **kwargs): ...
+
+@final
+class CType:
+    abi: Incomplete
+    args: Incomplete
+    cname: Incomplete
+    elements: Incomplete
+    ellipsis: Incomplete
+    fields: Incomplete
+    item: Incomplete
+    kind: Incomplete
+    length: Incomplete
+    relements: Incomplete
+    result: Incomplete
+    def __dir__(self): ...
+
+@final
+class Lib:
+    def __delattr__(self, name): ...
+    def __dir__(self): ...
+    def __setattr__(self, name, value): ...
```

Comment:
`__delattr__` and `__setattr__` have special meaning for type checkers if you override them. Both mypy and pyright will assume that any arbitrary attributes can be set on instances of the class if you override `__setattr__`. Pyright applies similar special-casing to `__delattr__`, though mypy doesn't yet.

Are these special semantics intended here? If not, it might be best to delete them:

```suggestion
    def __dir__(self): ...
```